src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx

```diff
@@ -230,6 +232,26 @@ export const TradeInput = ({
     return <TradeSettingsMenu isLoading={isLoading} isCompact={isCompact} />
   }, [isCompact, isLoading])
 
+  // Master effect syncing URL with state - note this is only done at trade input time
+  // That's the only place we want things to be in sync, other routes should be a redirect
```

Comment:
i.e no need for the ugly logic in `src/pages/Trade/tabs/TradeTab` anymore - `<MultiHopTrade />` directs the state sync from URL, and this directs the URL sync from state.
Routes other than input don't really have a notion of two-way sync and are really just a redirect to input.